### ISSUE TITLE
Cover COVER_OPENCLOSE_CMDS

### ISSUE THREAD

**Issue by waraileron** [rospogrigio/localtuya#144]
Thank you for the integration!

Currently I'm on the latest master branch
Device: Zemismart ZM25 with inverted control
DPS 1 for open, close, stop. DPS 2 for set position. DPS 3 for current position. Inverted checked.
Position updates correctly, open and stop works. 

open_close_stop does not allow "close" command to appear nor activated
editing the line `COVER_OPENCLOSE_CMDS = "open_close_stop"` shows that lowercase "close" messes with something out of my ability to debug. Variation of close command except full lowercase allows it to appear and be triggered but blind motor requires lowercase close.

Please let me know if more information is required, TIA.

**Comment by waraileron** [comment]
![image](https://user-images.githubusercontent.com/20854284/98605463-71cb3300-2320-11eb-9578-03351814607d.png)
![image](https://user-images.githubusercontent.com/20854284/98605502-87405d00-2320-11eb-9ab6-3b5cfbe09b5b.png)

**Comment by postlund** [comment]
@rospogrigio will have to look at this when he has time.

**Comment by rospogrigio** [comment]
@waraileron Sorry, I don't seem to understand clearly the problem, can you please explain better?
Can you also post your yaml config file?

**Comment by waraileron** [comment]
@rospogrigio Thank you for the reply, I went through integration configflow so there's no yaml.

Basically I have tested with on_off_stop, this shows the control perfectly fine however my blinds will not accept on/off as a command. However on open_close_stop, close would not appear anywhere in all lowercase form.

**Comment by rospogrigio** [comment]
You are just trying things by try-and-error, which is not the proper way to proceed.
Please provide a DPs dump (you can obtain it following this procedure: https://github.com/rospogrigio/localtuya/wiki/HOWTO-get-a-DPs-dump ), in the 3 situations (opening, closing, stopped). You can either use physical buttons if they are available, or use the Tuya app.
Let us know what comes out, bye...

**Comment by waraileron** [comment]
Well then, this is the dump
After pressing close on tuya app
`DEBUG:localtuya.pytuya:decrypted result='{"dps":{"1":"close","2":100,"3":100,"5":true,"7":"closing"}}'
AVAILABLE DPS ARE [{'1': 'close', '2': 100, '3': 100, '5': True, '7': 'closing'}]
INFO:localtuya:COMPLETE response from device <> [<>].

**** deviceInfo returned OK ****

TuyaDebug (Tuya DPs dump) [1.0.0]

Device <> at <> key <> protocol 3.3 dev_type type_0a:
    DPS [1] VALUE [close]
    DPS [2] VALUE [100]
    DPS [3] VALUE [100]
    DPS [5] VALUE [True]
    DPS [7] VALUE [closing]`

After pressing open
`DEBUG:localtuya.pytuya:decrypted result='{"dps":{"1":"open","2":100,"3":0,"5":true,"7":"closing","10":0}}'
AVAILABLE DPS ARE [{'1': 'open', '2': 100, '3': 0, '5': True, '7': 'closing', '10': 0}]
INFO:localtuya:COMPLETE response from device <> [<>].

**** deviceInfo returned OK ****

TuyaDebug (Tuya DPs dump) [1.0.0]

Device <> at <> key <> protocol 3.3 dev_type type_0a:
    DPS [1] VALUE [open]
    DPS [2] VALUE [100]
    DPS [3] VALUE [0]
    DPS [5] VALUE [True]
    DPS [7] VALUE [closing]
    DPS [10] VALUE [0]`

**Comment by waraileron** [comment]
I have seem to resolve the issue with the following step,
Press open/close then stop on tuya app

On observing the state tab, it would seem the state opening/closing was stuck previously somewhere during the set position process. For now I would need to play around to reproduce this issue again. Thank you for your help.

**Comment by waraileron** [comment]
I think I'm getting somewhere, state of cover (opening/closing) somehow gets disrupted somewhere from updating which stops the relevant control icon to appear.
3 identical blinds
![image](https://user-images.githubusercontent.com/20854284/98803908-6840e880-2450-11eb-90ce-4717e9b1325b.png)
![image](https://user-images.githubusercontent.com/20854284/98803962-7989f500-2450-11eb-93a4-62d1cf8afef2.png)
![image](https://user-images.githubusercontent.com/20854284/98803937-7131ba00-2450-11eb-8a54-763941f4da71.png)

**Comment by juslex** [comment]
hello, I have the same cover model, how do I fill in the DPS?  could you send me screenshots just like you did with the correct values ​​in the integration configuration?  Thank you very much

**Comment by waraileron** [comment]
> hello, I have the same cover model, how do I fill in the DPS? could you send me screenshots just like you did with the correct values ​​in the integration configuration? Thank you very much

![image](https://user-images.githubusercontent.com/20854284/98873483-6eb47c00-24b3-11eb-871f-83e850ae8b41.png)

**Comment by juslex** [comment]
> I think I'm getting somewhere, state of cover (opening/closing) somehow gets disrupted somewhere from updating which stops the relevant control icon to appear.
> 3 identical blinds
> ![image](https://user-images.githubusercontent.com/20854284/98804351-0af96700-2451-11eb-8ea8-30c95f77f249.png)
> ![image](https://user-images.githubusercontent.com/20854284/98804380-1482cf00-2451-11eb-9a41-572d57d3609e.png)
> ![image](https://user-images.githubusercontent.com/20854284/98804399-1a78b000-2451-11eb-88a4-7dc46c210655.png)
> 
> It could be opening smartlife/tuya app messes with the state or home assistant end, but setting the state on home assistant developer tools seems to have fixed the issue temporarily. Will observe for a little while.

Same here. after restarting the home assistant, the state of the curtain is lost and it is necessary to manually set it again

**Comment by waraileron** [comment]
Have since disabled def is_opening(self): and def is_closing(self):, it may be sequence of update that causes the state to be stuck in opening/closing state due to inverted positioning.
Only thing lost through disabling this is that you would not know remotely if the cover is opening/closing but that's fine since they open in about 20 seconds.

**Comment by juslex** [comment]
where do I find this option?

**Comment by waraileron** [comment]
in /config/custom_components/localtuya/cover.py lines 95-105 which i just comment out with #

**Comment by juslex** [comment]
Thank you so much. Everything is working perfectly. You are very kindly!

**Comment by juslex** [comment]
Dear,

Have you upgraded to version 3.1.0? Does everything still work?

**Comment by waraileron** [comment]
> Dear,
> 
> Have you upgraded to version 3.1.0? Does everything still work?

3.1.0 does not fix the issue, you would still need to comment out the opening and closing.

**Comment by rospogrigio** [comment]
@waraileron and @juslex , commenting those methods is not the proper way to fix these issues. Unfortunately I don't have much time currently to work on this, but it would be helpful if you sent a DPs dump of the device when you are in a wrong situation, so we can investigate the code. Thank you

**Comment by waraileron** [comment]
@rospogrigio no worries, this is a minor issue that is not a priority.
I've misread the dump required, please find attached log for: 1. opening on HA 2. closing on app 3. opening on HA 4. stop on HA. Device is stuck on closing.
[homeassistant.log](https://github.com/rospogrigio/localtuya/files/5613068/homeassistant.log)

**Comment by d3multimedia** [comment]
Somewhere the state of "closing" in cover.py is getting set, but not being changed afterword preventing the closing button.
If manually change state in the Developer Tools to something else, the down arrow becomes active and works, but regardless of which button is pressed disables the down arrow, and again set to "closing"
Not as familiar with python as I would like to try and fix it myself as I can't seem to find where the "closing" attribute is being set.

**Comment by WarlaxZ** [comment]
I have exactly the same issue. I've submitted a pull request which probably should fix it, but will need to wait on it getting merged. incredibly annoying as it makes the curtains basically useless, as they can only work once (without manually hacking it)
